trieste/bayesian_optimizer.py

```diff
@@ -255,7 +255,7 @@ def optimize(
                     model.update(dataset)
                     model.optimize(dataset)
 
-            except Exception as error:
+            except Exception as error:  # pylint: disable=broad-except
```

Comment:
Generally it's a bad idea to catch all `Exception`s, but it makes sense here, so I've disabled the warning just for this line.

Reply:
:)